Detectors/MUON/MCH/Raw/Decoder/src/DataDecoder.cxx

```diff
@@ -363,28 +382,18 @@ void DataDecoder::computeDigitsTime_(RawDigitVector& digits, SampaTimeFrameStart
   for (size_t di = 0; di < digits.size(); di++) {
     Digit& d = digits[di].digit;
     SampaInfo& info = digits[di].info;
-    std::optional<SampaTimeFrameStart>& tfStart = sampaTimeFrameStarts[info.id];
-
-    if (!tfStart) {
-      LOG(ERROR) << "\n[computeDigitsTime_] missing TimeFrame start for ID " << info.id << "  DS " << info.solar << ","
-                 << info.ds << " (J" << (info.ds / 5) + 1 << " B" << (info.ds % 5) << ")," << info.chip
-                 << "  pad " << d.getDetID() << "," << d.getPadID() << "  "
-                 << info.orbit << " " << info.tfTime << " " << info.bunchCrossing << " "
-                 << info.sampaTime << " " << info.getBXTime();
-      d.setTime(timeInvalid);
-      info.tfTime = timeInvalid;
-    } else {
-      int32_t tfTime = 0;
-      uint32_t bc = tfStart->mBunchCrossing;
-      uint32_t orbit = tfStart->mOrbit;
-      tfTime = DataDecoder::digitsTimeDiff(orbit, bc, info.orbit, info.getBXTime());
-      if (debug) {
-        std::cout << "\n[computeDigitsTime_] hit " << info.orbit << "," << info.getBXTime()
-                  << "    tfTime(1) " << orbit << "," << bc << "    diff " << tfTime << std::endl;
-      }
-      setDigitTime(d, tfTime);
-      info.tfTime = tfTime;
+    SampaTimeFrameStart& tfStart = sampaTimeFrameStart;
```

Comment:
this is a detail but why not using directly sampaTimeFrameStart instead of a copy?

Reply:
Yes, indeed.